hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/records/ApplicationHomeSubCluster.java

```diff
@@ -123,32 +125,35 @@ public static ApplicationHomeSubCluster newInstance(ApplicationId appId, long cr
 
   @Override
   public boolean equals(Object obj) {
+
     if (this == obj) {
       return true;
     }
-    if (obj == null) {
-      return false;
-    }
-    if (getClass() != obj.getClass()) {
-      return false;
-    }
-    ApplicationHomeSubCluster other = (ApplicationHomeSubCluster) obj;
-    if (!this.getApplicationId().equals(other.getApplicationId())) {
+
+    if (obj == null || getClass() != obj.getClass()) {
```

Comment:
Most other classes use instanceof

Reply:
Thanks for your suggestion, I will fix it.